package/gluon-hoodselector/luasrc/usr/sbin/hoodselector

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/lua
+
+local nixio = require('nixio')
+local uci = require('simple-uci').cursor()
+local hoodutil = require("hoodselector.util")
+
+-- PID file to ensure the hoodselector isn't running parallel
+local lockfile = '/var/lock/hoodselector.lock'
+local lockfd = nixio.open(lockfile, 'w', 'rw-------')
+
+if not lockfd:lock('tlock') then
+  io.stderr:write(string.format(
+    "Unable to lock file %s. Make sure there is no other instance of the hoodselector running.\n",
+    lockfile
+  ))
+  os.exit(1)
+end
+
+-- initialisation done
+--
+local function get_mesh_vpn_interface()
+  local ret = {}
+  if hoodutil.fastd_installed() then
+    local vpnifac = uci:get('fastd', 'mesh_vpn_backbone', 'net')
+    if vpnifac  ~= nil then
+      vpnifac = vpnifac:gsub("%_",'-')
+      table.insert(ret,vpnifac)
+    else
+      hoodutil.log("fastd uci config broken! abort...")
+      os.exit(1)
+    end
+  end
+  if hoodutil.tunneldigger_installed() then
+    local vpnifac = uci:get('tunneldigger', 'mesh_vpn', 'interface')
+    if vpnifac  ~= nil then
+      table.insert(ret,vpnifac)
+    else
+      hoodutil.log("tunneldigger uci config broken! abort...")
+      os.exit(1)
+    end
+  end
+  return ret
+end
+
+-- INITIALISE AND PREPARE DATA --
+-- read hoodfile...
+local jhood = hoodutil.get_domains()
+
+-- get default hood
+local defaultHood = hoodutil.getDefaultHood(jhood)
+
```

Comment:
we are parsing a lot of data just to realize that there is nothing to do. 
How could the be optimized for the most common path where nothing changed?